test/CMakeLists.txt

```diff
@@ -81,3 +81,30 @@ if (SINGULARITY_BUILD_PYTHON AND SINGULARITY_TEST_PYTHON)
            COMMAND ${Python_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/python_bindings.py -v)
   set_tests_properties(PythonBindings PROPERTIES ENVIRONMENT "${PYTHON_TEST_ENVIRONMENT}")
 endif()
+
+add_executable(profile_eos profile_eos.cpp)
+target_link_libraries(profile_eos singularity-eos::singularity-eos)
+
+if(SINGULARITY_USE_EOSPAC AND SINGULARITY_TEST_SESAME AND NOT SINGULARITY_USE_CUDA)
+  set(METHODS_TO_TEST
+    TemperatureFromDensityInternalEnergy
+    PressureFromDensityTemperature
+    InternalEnergyFromDensityTemperature
+    PressureFromDensityInternalEnergy
+    SpecificHeatFromDensityTemperature
+    SpecificHeatFromDensityInternalEnergy
+    BulkModulusFromDensityTemperature
+    BulkModulusFromDensityInternalEnergy
+    GruneisenParamFromDensityTemperature
+    GruneisenParamFromDensityInternalEnergy
+  )
+
+  foreach(MUT IN LISTS METHODS_TO_TEST)
+    add_executable(profile_eospac_${MUT} profile_eospac.cpp)
+    target_compile_definitions(profile_eospac_${MUT} PUBLIC METHOD_UNDER_TEST=${MUT})
+    target_link_libraries(profile_eospac_${MUT} singularity-eos::singularity-eos)
+
+    add_test(NAME Profile_EOSPAC_${MUT}
+             COMMAND profile_eospac_${MUT} 100 16)
+  endforeach()
+endif()
```

Comment:
Thanks again for adding this profiling test! This is excellent and we need to do more of this!